lib/manager/argocd/extract.ts

```diff
@@ -0,0 +1,61 @@
+import { safeLoadAll } from 'js-yaml';
+import * as gitTags from '../../datasource/git-tags';
+import * as helm from '../../datasource/helm';
+import type { ExtractConfig, PackageDependency, PackageFile } from '../types';
+import type { ApplicationDefinition } from './types';
+import { fileTestRegex } from './util';
+
+function loadYaml(content: string): ApplicationDefinition[] {
+  return safeLoadAll(content);
+}
+function createDependency(
+  definition: ApplicationDefinition
+): PackageDependency {
+  let result: PackageDependency;
+
+  const source = definition.spec.source;
+
+  if (source == null) {
+    return null;
+  }
+
+  // a chart variable is defined this is helm declaration
+  if (source.chart) {
+    result = {
+      depName: source.chart,
+      registryUrls: [source.repoURL],
+      currentValue: source.targetRevision,
+      datasource: helm.id,
+    };
+  } else {
+    result = {
+      depName: source.repoURL,
+      currentValue: source.targetRevision,
+      datasource: gitTags.id,
+    };
+  }
+  return result;
+}
+
+export function extractPackageFile(
+  content: string,
+  fileName: string,
+  config?: ExtractConfig
+): PackageFile | null {
+  // check for argo reference. API version for the kind attribute is used
+  if (fileTestRegex.test(content) === false) {
```

Comment:
Nice, so we can save parsing phase if we know this is no argocd  document. 👍 

@rarkins I think we can enable it by default for all yaml files as this regex will ignore files which do not have argocd content.

Reply:
My main concern is that repos may have hundreds of YAML files while only a very small fraction would be argo or even use argo.

Is there any naming convention which would cover maybe 80%+ of the files e.g. `/argo.*\.yaml$/i` ?

If so then I'd prefer that as the default. But if not, it could be ok to default to all `.yaml`

Reply:
No, there isn't such conventions.    
Basically this is updating Kubernetes manifests which contain ArgoCD CR manifests. Because of that fact this every `.yaml` could contain an ArgoCD manifest.

Reply:
OK, let's see how it goes then. I think we may read detected files into memory in parallel, so it could increase memory use by a lot if there are large YAML files.